Tasks/Petrov/Task7/task7_5_local_cache.py

```diff
@@ -0,0 +1,51 @@
+from decimal import Decimal, InvalidOperation
+
+# dictionary for all functions, that will be cached
+caches = {}
+
+
+def caching(func):
+    """Decorator function that caches the return value of a function
+    with a dictionary.
+
+    func
+        function that will be decorated"""
+
+    # Creating individual dictionary for certain function,
+    # that will be cached
+    caches[func.__name__] = {}
+    cache_current = caches[func.__name__]
```

Comment:
you absolutely correct to choose such an approach in terms of different functions caching